### DISCUSSION TITLE
VPN connection with Crowdin API

### DISCUSSION THREAD

**Discussion by zakyen** [crowdin/crowdin-api-client-python#66] [Q&A] [answered]
Good morning,

We are using a VPN in our work environment and our Crowdin API requests are failing because it needs to use a certificate when making https requests.

The requests library that the implementation uses, supports the use of certificates in the parameters of the different methods but the python client implementation of Crowdin overrides this. Is there any way to pass the certificate to the requests we make to the Crowdin API?

**Explanation**:
The function **list_projects** located in crowdin_api > api_resources > projects > resource calls the function **_get_entire_data** located in crowdin_api > api_resources > abstract > resources and this one use the **requester.request** function that is an implementation of the requester.py located in crowdin_api. This call is consuming the library requests https://pypi.org/project/requests/
```
self.requester.request(
                method=method,
                path=path,
                params=params,
            )
```
The request function inside the requests library looks like this :
```
 def request(
        self,
        method,
        url,
        params=None,
        data=None,
        headers=None,
        cookies=None,
        files=None,
        auth=None,
        timeout=None,
        allow_redirects=True,
        proxies=None,
        hooks=None,
        stream=None,
        verify=None,
        cert=None,
        json=None,
    ): 
```
But the call made to this function by your implementation does not accept the additional verify parameter that is needed to pass a certificate file to validate the https connection to a VPN. 
Is there any solution to use certificates in HTTPS calls? This was just an example of how to retrieve the projects, but the other API calls follow the same implementation.
Thanks for your attention

**Comment by andrii-bodnar** [comment]
Will be fixed via https://github.com/crowdin/crowdin-api-client-python/pull/67

**Comment by andrii-bodnar** [comment, marked as answer]
Hi @zakyen,

available in the [1.9.0](https://github.com/crowdin/crowdin-api-client-python/releases/tag/v1.9.0) version 🚀

**Comment by zakyen** [comment]
Hi @andrii-bodnar  I will try to test it the next week, thanks a lot.
